tests/src/test/scala/ironrust/FullIntegrationTest.scala

```diff
@@ -198,7 +198,7 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
       val jwt = JwtHelper.generateValidJwt(primaryTestUserId.getId)
 
       // a second device
-      Try(IronOxide.generateNewDevice(jwt, primaryTestUserPassword, DeviceCreateOpts.create(null), null)).toEither
+      Try(IronOxide.generateNewDevice(jwt, primaryTestUserPassword, new DeviceCreateOpts(null), null)).toEither
```

Comment:
There's not much we can do to prevent someone from calling `new DeviceCreateOpts(null)`, but I think it's marginally better to show this being called as `new DeviceCreateOpts()`

Reply:
I don't think that calling `new DeviceCreateOpts(null)` is something we need to discourage? If for some reason we decided to change `DeviceCreate::default()` in the future to set the name to "undefined", then passing null would be the only way to set the name to `None`. So I think calling it like that is a good way to ensure that it's set as desired.
Thoughts?

Reply:
Maybe I'm missing your point... Usually in Java, calling a constructor with no arguments gives you the defaults, whatever those may be (thus the name default constructor). If we changing the underlying default to "undefined", this default will also change. 

Maybe what I'm realizing, and suggesting here, is that if we want to provide default behavior, the default constructor is the best way to do that in Java. That doesn't mean we can't provide other constructors, and it doesn't mean that someone couldn't reproduce the default behavior with another constructor. I didn't go back and look, but maybe what I'm suggesting here isn't consistent with what we do other places. If that's the case I would suggest we have a larger discussion outside of this PR.